common/statistics/src/clients/gateway_conn_statistics.rs

```diff
@@ -0,0 +1,77 @@
+//! # Gateway Connection statistics
+//!
+//! Metrics collected by the client while establishing and maintaining connections to the gateway.
+
+use super::ClientStatsEvents;
+use std::collections::VecDeque;
+
+use nym_metrics::{inc, inc_by};
+use serde::{Deserialize, Serialize};
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub(crate) struct GatewayStats {
+    // Sent
+    real_packets_sent: u64,
+    real_packets_sent_size: usize,
+
+    /// failed connection statistics
+    failures: VecDeque<()>, // TODO
+}
+
+impl GatewayStats {
+    fn handle(&mut self, event: GatewayStatsEvent) {
+        match event {
+            GatewayStatsEvent::RealPacketSent(packet_size) => {
+                self.real_packets_sent += 1;
+                self.real_packets_sent_size += packet_size;
+                inc!("real_packets_sent");
+                inc_by!("real_packets_sent_size", packet_size);
+            }
+        }
+    }
+
+    fn summary(&self) -> (String, String) {
```

Comment:
can we use more strongly-typed return type? what's this tuple meant to represent? the second string is empty, how is anyone meant to know what it's meant to be?

Reply:
Pinging @jmwample for this one

Reply:
For now these are place-holders and internal to the type which remain as part of the skeleton from the `packet-statistics` blueprint. The statistics collected here will be expanded and the serialization will be more structured. The primary goal was to get the structures in place for aggregated stats collection from multiple sources.